### PR TITLE
Added support for nested list types

### PR DESCRIPTION
**Pull request by Akshay2402**
Fixes: [10070](https://github.com/postmanlabs/postman-app-support/issues/10070)

### CODE REVIEW

File: lib/assets/gql-generator.js [hunk 1]
```diff
@@ -125,9 +134,27 @@ function resolveVariableType (type, gqlSchema, stack = 0, stackLimit = 5) {
...
     return fieldObj;
   }
-  return type.toString().includes('[') ?
-    [getDefaultForType(argType.toString())] :
-    getDefaultForType(argType.toString());
+
+  // If type is an array, get the default value for the type and return the desired depth nested array
+  if (type.toString().includes('[')) {
+    const getArrayDepth = (value) => {
+        return (value.match(/\[/g)).length;
```
Comment (VShingala) [inline lib/assets/gql-generator.js:141]: @Akshay2402 This should work for expected cases but not sure what's the standard for edge cases like below.

```
[[String]
[String
String]
[Stri[ng]
```

**Comment by VShingala** [comment]
@Akshay2402 PR should be based from `develop` and merged to it as well. Let's rebase it.

**Comment by Akshay2402** [comment]
> @Akshay2402 PR should be based from `develop` and merged to it as well. Let's rebase it.

Done

File: lib/assets/gql-generator.js [hunk 2]
```diff
@@ -23,11 +32,37 @@ var graphql = require('graphql'),
...
+            splitString += identifier;
+
+            ArrayDepth--;
+          }
+
+          return splitString;
+        },
+        depth = getDepthOfArray(type);
+
+      type = type.split(getSplitString(depth, '['))[1].split(getSplitString(depth, ']'))[0];
```
Comment (VShingala) [inline lib/assets/gql-generator.js:65]: @Akshay2402 Try to simplify this. i.e. we're nesting many calls and references in one sentence here. We can do it in steps.

**Review by VShingala — APPROVED** [review]
PR looks good, added one comment for improvement..

File: lib/assets/gql-generator.js (hunk 2 above)
Comment (Akshay2402) [reply lib/assets/gql-generator.js]: Done

**Pull request merged**